src/ol/interaction/translateinteraction.js

```diff
@@ -0,0 +1,126 @@
+goog.provide('ol.interaction.Translate');
+
+goog.require('ol.interaction.Pointer');
+
+
+
+/**
+ * @classdesc
+ * Interaction for translating (moving) features.
+ *
+ * @constructor
+ * @extends {ol.interaction.Pointer}
+ * @param {olx.interaction.TranslateOptions} options Options.
+ * @api
+ */
+ol.interaction.Translate = function(options)
+    {
+  goog.base(this, {
+    handleDownEvent: ol.interaction.Translate.handleDownEvent_,
+    handleDragEvent: ol.interaction.Translate.handleDragEvent_,
+    handleUpEvent: ol.interaction.Translate.handleUpEvent_
+  });
+
+  /**
+    * The last position we translated to.
+    * @type {ol.Coordinate}
+    * @private
+    */
+  this.lastCoordinate_ = null;
+
+
+  /**
+   * @type {ol.Collection.<ol.Feature>}
+   * @private
+   */
+  this.features_ = options.features;
+};
+goog.inherits(ol.interaction.Translate, ol.interaction.Pointer);
+
+
+/**
+ * @inheritDoc
+ */
+ol.interaction.Translate.prototype.setMap = function(map) {
+  goog.base(this, 'setMap', map);
+};
+
+
+/**
+ * @param {ol.MapBrowserPointerEvent} event Event.
+ * @return {boolean} Start drag sequence?
+ * @this {ol.interaction.Translate}
+ * @private
+ */
+ol.interaction.Translate.handleDownEvent_ = function(event) {
+  if (goog.isNull(this.lastCoordinate_) &&
+      this.intersectsFeature_(event.pixel, event.map)) {
+    this.lastCoordinate_ = event.coordinate;
+    return true;
+  }
+  return false;
+};
+
+
+/**
+ * @param {ol.MapBrowserPointerEvent} event Event.
+ * @return {boolean} Stop drag sequence?
+ * @this {ol.interaction.Translate}
+ * @private
+ */
+ol.interaction.Translate.handleUpEvent_ = function(event) {
+  if (!goog.isNull(this.lastCoordinate_)) {
+    this.lastCoordinate_ = null;
+    return true;
+  }
+  return false;
+};
+
+
+/**
+ * @param {ol.MapBrowserPointerEvent} event Event.
+ * @this {ol.interaction.Translate}
+ * @private
+ */
+ol.interaction.Translate.handleDragEvent_ = function(event) {
+  if (!goog.isNull(this.lastCoordinate_)) {
+    var newCoordinate = event.coordinate;
+    var deltaX = newCoordinate[0] - this.lastCoordinate_[0];
+    var deltaY = newCoordinate[1] - this.lastCoordinate_[1];
+
+    this.features_.forEach(function(feature)
+        {
+          var geom = feature.getGeometry();
+          geom.translate(deltaX, deltaY);
+          feature.setGeometry(geom);
+        });
+
+    this.lastCoordinate_ = newCoordinate;
+  }
+};
+
+
+/**
+ * Tests to see if the given coordinates intersects any of our selected
+ * features.
+ * @param {ol.Pixel} pixel Pixel coordinate to test for intersection.
+ * @param {ol.Map} map Map to test the intersection on.
+ * @return {boolean} Returns `true` if the coordinate intersects a
+ * selected feature, otherwise `false`.
+ * @private
+ */
+ol.interaction.Translate.prototype.intersectsFeature_ = function(pixel, map) {
```

Comment:
I'm not sure what other term you would use? We're essentially doing an intersection between the point that was clicked, and all features on the map, both in a mathematical sense, and a geometrical sense.

I'd be happy to use a better term, I just can't think of one. :)